sympy/physics/units/systems/si.py

```diff
@@ -253,6 +253,10 @@
 SI.set_quantity_dimension(astronomical_unit, length)
 SI.set_quantity_scale_factor(astronomical_unit, 149597870691*meter)
 
+SI.set_quantity_dimension(angstrom, length)
+SI.set_quantity_scale_factor(angstrom, 1*meter/10**10)
```

Comment:
This only sets the scale factor for SI units. The scale factor should be set for all unit systems.

Reply:
> This only sets the scale factor for SI units. The scale factor should be set for all unit systems.

But when I convert `angstrom` into `inches`or in `foot` or`mm` it gives correct results in my local branch.

Reply:
`convert_to` uses SI units by default. Try using cgs units.

Reply:
you are saying to add these things or anything else 
```
SI.set_quantity_scale_factor(angstrom, 1*centimeter/10**8)
SI.set_quantity_scale_factor(angstrom, 1*millimeter/10**7)
SI.set_quantity_scale_factor(angstrom, 3.9370078740157*inch/10**9)
SI.set_quantity_scale_factor(angstrom, 3.280839895*foot/10**10)
```

Reply:
No. The scale factor should only be defined once, globally for all unit systems.

Reply:
As you said before that scale factor should be set for all unit systems but now...
Could you pls give an example so that I understand

Reply:
Only `angstrom.set_global_relative_scale_factor( ... )` should be called. `SI.set_quantity_scale_factor( ... )` only sets the scale factor for SI units.

```python
In [1]: from sympy.physics.units.systems import SI

In [2]: from sympy.physics.units.systems.cgs import cgs_gauss

In [3]: from sympy import *; from sympy.physics.units import *

In [4]: angstrom = angstroms = Quantity("angstrom", latex_repr=r'\r{A}')

In [5]: SI.set_quantity_scale_factor(angstrom, 1*meter/10**10)

In [7]: SI.set_quantity_dimension(angstrom, length)

In [8]: convert_to(angstrom, meter, SI)
Out[8]: meter/10000000000

In [9]: convert_to(angstrom, meter, cgs_gauss)
Out[9]: angstrom
```

As you can see in this example, output 9 does not work, because the scale factor has only been set for SI units, so calling `convert_to` in `cgs_gauss` units does not work. Therefore, you should only use the global set factor in this case (there are other cases where the scale factor is different depending on the chosen unit system, in those cases it's correct to set the quantity scale factor inside the unit system object).

Reply:
done!